citeproc-java/src/test/java/de/undercouch/citeproc/CSLTest.java

```diff
@@ -553,4 +557,79 @@ public void dependentStyle() throws Exception {
+                "        \"collection-number\": \"Series Number\",\n" +
+                "        \"publisher\": \"Publisher\",\n" +
+                "        \"publisher-place\": \"Place\",\n" +
+                "        \"volume\": \"Volume\",\n" +
+                "        \"number-of-volumes\": \"# of Volumes\",\n" +
+                "        \"number-of-pages\": \"# of Pages\",\n" +
+                "        \"edition\": \"Edition\",\n" +
+                "        \"source\": \"Library Catalog\",\n" +
+                "        \"archive\": \"Archive\",\n" +
+                "        \"archive_location\": \"Loc. in Archive\",\n" +
+                "        \"event-place\": \"Place\",\n" +
+                "        \"abstract\": \"Abstract\",\n" +
+                "        \"URL\": \"URL\",\n" +
+                "        \"ISBN\": \"ISBN\",\n" +
+                "        \"call-number\": \"Call Number\",\n" +
+                "        \"note\": \"Extra\",\n" +
+                "        \"title-short\": \"Short Title\",\n" +
+                "        \"language\": \"Language\",\n" +
+                "        \"author\": [\n" +
+                "            {\n" +
+                "                \"literal\": \"Author_Fullname\"\n" +
+                "            }\n" +
+                "        ],\n" +
+                "        \"issued\": {\n" +
+                "            \"literal\": \"Date\"\n" +
+                "        }\n" +
+                "    }\n" +
+                "]";
+        InputStream is = new ByteArrayInputStream(BOOK_WITH_UNEXPECTED_FIELDS_JSON.getBytes(StandardCharsets.UTF_8));
+        ItemDataProvider provider = new BibliographyFileReader().readBibliographyFile(is, BibliographyFileReader.FileFormat.JSON_ARRAY);
+        CSLItemData item = provider.retrieveItem(provider.getIds()[0]);
+
+        final String EXPECTED = "Author_Fullname. (Date). Book_Title (Edition). Retrieved from URL";
+
+        assertEquals(EXPECTED, CSL.makeAdhocBibliography("apa", "text", item).makeString().trim());
+    }
+
+    @Test
+    public void shouldRenderAnEntryWithUnexpectedFields_FromBIBTEX_Correctly() throws IOException {
+        final String BOOK_WITH_UNEXPECTED_FIELDS_BIBTEX =
+                "@book{author_fullname_book_title_nodate,\n" +
+                "    address = {Place},\n" +
+                "    edition = {Edition},\n" +
+                "    series = {Series},\n" +
+                "    title = {Book\\_Title},\n" +
+                "    volume = {Volume},\n" +
+                "    copyright = {Rights},\n" +
+                "    isbn = {ISBN},\n" +
+                "    shorttitle = {Short {Title}},\n" +
+                "    url = {URL},\n" +
+                "    abstract = {Abstract},\n" +
+                "    language = {Language},\n" +
+                "    number = {Series Number},\n" +
+                "    publisher = {Publisher},\n" +
+                "    author = {{Author\\_Fullname}},\n" +
```

Comment:
@michel-kraemer Probably not a good idea for a name: the `de.undercouch.citeproc.bibtex.NameParser` doesn't like `_` and trims the `_Fullname` from the name. Not sure if you'd like to review the parser, or if I just should choose a more beautiful name